sisl/io/sile.py

```diff
@@ -546,6 +547,50 @@ def pre_open(self, *args, **kwargs):
     return _wrapper
 
 
+def sile_read_multiple(all=False, start=0, step=None):
+    """ Method decorator for doing multiple reads
+
+    Parameters
+    ----------
+    all : bool, optional
+       read all entries (takes precedence over start/step)
+    start : int, optional
+       position of first read
+    step : int, optional
+       number of steps to take (if any at all)
+
+    Returns
+    -------
+    single entry or list from multiple reads
+    """
+    def decorator(func):
+        @wraps(func)
+        def multiple(self, *args, all=all, start=start, step=step, **kwargs):
+            if all is True:
+                start, step = 0, 1
+            R, r = [], True
+            if hasattr(self, "fh"):
+                if step is None:
+                    for s in range(start):
+                        func(self, *args, **kwargs)
+                    return func(self, *args, **kwargs)
+                while r is not None:
+                    R.append(r)
+                    r = func(self, *args, **kwargs)
+                return R[start+1::step]
```

Comment:
shouldn't this be `R[start::step]`? otherwise you'll skip `start` element?

Reply:
It's because the first entry is `True` because I append to `R` before reading the first `r`. Maybe a sloppy way to do it.

Reply:
ah, I see... That didn't catch my eye at first